File: tests/fixtures/search/go/models/order.go
package models

type Order struct {
	ID    int64
	Items []string
}

type OrderID = int64

const DefaultCurrency = "USD"

func MakeOrder(id int64) *Order {
	return &Order{ID: id}
}

func (o *Order) Total() int {
	return len(o.Items)
}

func DuplicateOrderID() OrderID {
	return 0
}

File: tests/fixtures/search/rs/src/models/order.rs
pub struct Order<T> {
    pub id: u64,
    pub items: Vec<T>,
}

impl<T> Order<T> {
    pub fn total(&self) -> u64 {
        self.items.len() as u64
    }
}

pub type OrderId = u64;

pub fn make_order<T>(items: Vec<T>) -> Order<T> {
    Order { id: 0, items }
}